qa/rolling-upgrade/src/test/java/org/opensearch/upgrades/IndexingIT.java

```diff
@@ -251,6 +269,10 @@ public void testIndexingWithSegRep() throws Exception {
                     .put(IndexMetadata.INDEX_NUMBER_OF_SHARDS_SETTING.getKey(), shardCount)
                     .put(IndexMetadata.INDEX_NUMBER_OF_REPLICAS_SETTING.getKey(), replicaCount)
                     .put(IndexMetadata.SETTING_REPLICATION_TYPE, ReplicationType.SEGMENT)
+                    .put(
+                        EngineConfig.INDEX_CODEC_SETTING.getKey(),
+                        randomFrom(CodecService.DEFAULT_CODEC, CodecService.BEST_COMPRESSION_CODEC, CodecService.LUCENE_DEFAULT_CODEC)
```

Comment:
Are we not checking the 9.x line of codecs? The ones that support write.

Reply:
Thanks @Poojita-Raj for the review. 
The bwc tests creates the scenario where primary and replica shards use different codecs and is the intention of this test. Using an older codec with writer better suits an integration test but previously @mch2 suggested to remove it as we were relying on reading the library directory. Discusion captured [here](https://github.com/opensearch-project/OpenSearch/pull/7903#discussion_r1220368461)